test/controllers/update/update.review.overseas.entity.information.controller.spec.ts

```diff
@@ -0,0 +1,93 @@
+jest.mock("ioredis");
+jest.mock("../../../src/utils/logger");
+jest.mock('../../../src/middleware/authentication.middleware');
+jest.mock('../../../src/middleware/company.authentication.middleware');
+jest.mock('../../../src/utils/application.data');
+jest.mock('../../../src/middleware/service.availability.middleware');
+jest.mock('../../../src/middleware/navigation/update/has.overseas.entity.middleware');
+
+import { NextFunction, Request, Response } from "express";
+import { beforeEach, expect, jest, test, describe } from "@jest/globals";
+import request from "supertest";
+import {
+  UPDATE_REVIEW_OVERSEAS_ENTITY_INFORMATION_PAGE,
+  UPDATE_REVIEW_OVERSEAS_ENTITY_INFORMATION_URL,
+  OVERSEAS_ENTITY_REVIEW_PAGE
+} from "../../../src/config";
+import app from "../../../src/app";
+import {
+  APPLICATION_DATA_MOCK
+} from '../../__mocks__/session.mock';
+import {
+  PAGE_TITLE_ERROR,
+  ANY_MESSAGE_ERROR,
+  SERVICE_UNAVAILABLE,
+  UPDATE_REVIEW_OVERSEAS_ENTITY_INFORMATION_PAGE_TITLE
+} from "../../__mocks__/text.mock";
+import { getApplicationData } from "../../../src/utils/application.data";
+import { authentication } from "../../../src/middleware/authentication.middleware";
+import { companyAuthentication } from "../../../src/middleware/company.authentication.middleware";
+import { serviceAvailabilityMiddleware } from "../../../src/middleware/service.availability.middleware";
+import { logger } from "../../../src/utils/logger";
+import { hasOverseasEntity } from "../../../src/middleware/navigation/update/has.overseas.entity.middleware";
+
+const mockHasOverseasEntity = hasOverseasEntity as jest.Mock;
+mockHasOverseasEntity.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
```

Comment:
this mock will have to change for your test then if your navigation check is to change.